### PR TITLE
Fix binlog artifact upload

### PR DESCRIPTION
**Pull request by weshaggard**
@heaths I was trying to set System.Debug=true on a pipeline and keep hitting failures around the binlog stuff. I don't think this has ever worked I've fixed this so that the logs get generated and uploaded. I also noticed that the binlog only gets uploaded for the tests today if you want the build and packaging we still need to add a similar upload step in ci.yml.

### CODE REVIEW

File: eng/pipelines/templates/jobs/ci.tests.yml [hunk 1]
```diff
@@ -117,9 +117,9 @@ jobs:
           mergeTestResults: true
       - template: /eng/common/pipelines/templates/steps/publish-artifact.yml
         parameters:
-          ArtifactPath: $(Build.ArtifactStagingDirectory)/*.binlog
-          ArtifactName: logs
-          CustomCondition: eq(variables['System.Debug'], 'true')
+          ArtifactPath: $(Build.ArtifactStagingDirectory)/test.binlog
+          ArtifactName: binlog_$(Agent.JobName)
```
Comment (heaths) [inline eng/pipelines/templates/jobs/ci.tests.yml:121]: Thanks. I swear I tested this, but maybe the conditional issue you fixed in the other file (`System.Debug` was documented to be available, but I didn't realize setting it queue time doesn't just update that globally scoped variable) and don't remember any conflicts. Besides, I incorrectly remembered AzDO can't just append to artifacts like GitHub Actions can...and I've been writing a heck of a lot more GitHub Actions the past few years.

**Review by heaths — APPROVED** [review]

**Comment by weshaggard** [comment]
The mac failure is unrelated to my changes so overriding that check.

**Comment by dw511214992** [comment]
aaa

**Comment by dw511214992** [comment]
aaa

**Pull request merged**